lib/internal/Magento/Framework/View/Layout/GeneratorPool.php

```diff
@@ -77,6 +77,10 @@ public function process(Reader\Context $readerContext, Generator\Context $genera
     {
         $this->buildStructure($readerContext->getScheduledStructure(), $generatorContext->getStructure());
         foreach ($this->generators as $generator) {
+            if (!$this->validateGenerator($generator)) {
+                continue;
```

Comment:
> Fixed Issues (if relevant)
> It fixes no issues.

What kind of issue it fixes then? What happened before if corresponding interface is not implemented?

Reply:
Hi @orlangur.

If any 3rd party module overrides or include a generator in GeneratorPool and doesn't declare the methods getType() and process() the user will be given an exception. That's why I think it's a good idea to force the implementation of `GeneratorInterface`.

Examples:

`Call to undefined method Any\Class::getType()`
`Call to undefined method Magento\Framework\View\Page\Config\Generator\Head::process()`

Reply:
@tiagosampaio so user already receives an exception and by this change you actually hide it. What should really be done is throwing exception in case of incorrect interface and only in `addGenerators` method.

Reply:
@orlangur so I think it's a better idea to throw a specific error to guide the user better to solve this problem. What do you think?

Reply:
@orlangur you're right. I'll do it.